core/arch/arm/include/mm/core_mmu.h

```diff
@@ -122,9 +122,12 @@ enum teecore_memtypes {
 	MEM_AREA_MAXTYPE
 };
 
-/* Default NSec shared memory allocated from NSec world */
-extern unsigned long default_nsec_shm_paddr;
```

Comment:
Maybe, my intention here was to make it similar to the tee_mm's we have in tee_mm.c. When we are dealing with ddr on secure side we're using `tee_mm_sec_ddr`. So I thought having the same for public (but without the "_mm_", since it's not handle in there directly) would be more appropriate.

The thing is that we have so many ways in the code to talk about memory and I think that is very confusing. I would like to go over all those functions, variables, enums, cfg's etc to see if you can make them more common. A quick grep and sort and pick a few names (there's plenty more) containing ddr, nsec, sec, ram, mem gives:
```
tee_mm_pub_ddr
tee_mm_sec_ddr
CFG_DDR_START
CFG_DDR_TEETZ_RESERVED_START
CFG_DDR_TEETZ_RESERVED_SIZE
DDR_PHYS_START
DDR_SIZE
TA_FLAG_EXEC_DDR
CFG_TA_RAM_SIZE
TZDRAM_BASE
CFG_DDR_SIZE
CFG_DDR_ARMTZ_ONLY_START
CFG_DDR_ARMTZ_ONLY_SIZE
CFG_TEE_RAM_START
sec_ddr_attr
teecore_init_pub_ram
CORE_MEM_NSEC_SHM
MEM_AREA_NSEC_SHM
MEM_AREA_TEE_RAM,                                                                                                                                                        
MEM_AREA_TEE_COHERENT,                                                                                                                                                   
MEM_AREA_TA_RAM,                                                                                                                                                         
MEM_AREA_NSEC_SHM,                                                                                                                                                       
MEM_AREA_KEYVAULT,                                                                                                                                                       
MEM_AREA_IO_SEC,                                                                                                                                                         
MEM_AREA_IO_NSEC,                                                                                                                                                        
MEM_AREA_TA_VASPACE
map_nsec_shm
```
I know that doing such a comparison is a little like mixing apples and oranges. But they are related in some sense and it would be nice to have a common naming schema/convention.

Having that said, I could skip the renaming in this patch and then attack it later on, in a separate refactoring patch if that is preferred?